### ISSUE TITLE
fix the workflow for notifications

### ISSUE THREAD

**Issue by cjdcordeiro** [nuvla/api-server#336]
make it possible to dismiss a notification

**Comment by konstan** [comment]
From the design of notifications for new images
https://github.com/nuvla/nuvla/wiki/Notification-on-New-Image-Version-in-Docker-Registry
the following applies in general

a. Users should be able to opt-in/-out of subscription to the notifications
c. Users can apply the following actions on the notification:
  1. dismiss current notification,
  2. defer current notification,
  3. dismiss and reschedule the notification.

**Comment by cjdcordeiro** [comment]
I agree with that. How far is it from the current implementation? we need a subscription workflow...

also, 2. and 3., what's the difference? Cause defer means dismiss for now and re-schedule for 30m from now

**Comment by konstan** [comment]
Yes, subscription workflow is needed.

Dismiss - delete. Remove from UI and delete the associated notification resource.
Defer - temporarily hide the notification. Remove the notification from UI. When you "defer" you do not delete the notification resource, instead you set `hide-until`. 
Each notification has a hash. If you defer a notification for 1h, but the subscription is set to 10min, the notification creation service will still attempt to compose a notification, compute its hash, but it will not create a new one if the same one (data-wise) already exists. So, it is guaranteed you don't receive the same notification (data-wise) before 1h has passed. After 1h Notifier will tigger the notification (deleting `hide-until`).

3. dismiss and reschedule the notification.
The notification gets deleted and the schedule in the X subscription gets updated with the new value the user requested.

**Comment by cjdcordeiro** [comment]
hum it's a bit confusing...cause functionally, it's the same...whether you re-schedule it or hide it, the user will eventually get notified again in X time...

**Comment by konstan** [comment]
Because you don't want to change 10min schedule for the notifications of the type X. You just want to temporarily defer the particular notification, i.e. to be reminded to do something about a particular instance of X-thing.
